pkg/controllers/workgenerator/controller.go

```diff
@@ -608,6 +604,40 @@ func getWorkNamePrefixFromSnapshotName(resourceSnapshot *fleetv1beta1.ClusterRes
 	return fmt.Sprintf(fleetv1beta1.WorkNameWithSubindexFmt, crpName, subIndexVal), nil
 }
 
+// setBindingStatus sets the binding status based on the works
+func setBindingStatus(works map[string]*fleetv1beta1.Work, resourceBinding *fleetv1beta1.ClusterResourceBinding) {
+	bindingRef := klog.KRef(resourceBinding.Namespace, resourceBinding.Name)
+	// try to gather the resource binding applied status if we didn't update any associated work spec this time
```

Comment:
need to reset the FailedPlacements.

resourceBinding.Status.FailedPlacements = nil